homeassistant/components/shelly/utils.py

```diff
@@ -353,18 +354,33 @@ def is_rpc_momentary_input(
     return cast(bool, config[key]["type"] == "button")
 
 
-def is_block_channel_type_light(settings: dict[str, Any], channel: int) -> bool:
+def is_block_channel_type_light(settings: dict[str, Any], channel: str) -> bool:
     """Return true if block channel appliance type is set to light."""
-    app_type = settings["relays"][channel].get("appliance_type")
+    app_type = settings["relays"][int(channel)].get("appliance_type")
     return app_type is not None and app_type.lower().startswith("light")
 
 
-def is_rpc_channel_type_light(config: dict[str, Any], channel: int) -> bool:
+def is_block_exclude_from_relay(settings: dict[str, Any], block: Block) -> bool:
+    """Return true if block should be excludeed from switch platform."""
+
+    if settings.get("mode") == "roller":
+        return True
+
+    if block.channel is None:
+        return False
```

Comment:
Looks like we are missing coverage for this line

Reply:
True, but is not changed by this PR ;-)

Reply:
Is it possible this is `None` here?

If its only for type checking you could write it as 

```
    if TYPE_CHECKING:
        assert block.channel is not None
```

Reply:
Yes, I think is only for that.

We should if TYPE_CHECKING in many other places. Currently there are merely asserts.

Reply:
If its only for type checking than guard with `if TYPE_CHECKING:`, if its possible we might actually get into that state and it would be a bug in the integration if were were, than use a bare assert

Reply:
Done